src/hotspot/os/linux/cgroupV1Subsystem_linux.cpp

```diff
@@ -145,6 +145,13 @@ jlong CgroupV1Subsystem::memory_and_swap_limit_in_bytes() {
     }
     return (jlong)-1;
   } else {
+    GET_CONTAINER_INFO(julong, _memory->controller(), "/memory.swappiness",
+                       "Swappiness is: " JULONG_FORMAT, JULONG_FORMAT, swappiness);
+    if (swappiness == 0) {
+      jlong memlimit = read_memory_limit_in_bytes();
+      log_trace(os, container)("Memory and Swap Limit has been reset to " JULONG_FORMAT " because swappiness is 0", memlimit);
+      return memlimit;
+    }
```

Comment:
It now occurs to me that we'd need the same treatment of this on line `143`. Perhaps extract a function and call it in both places?